kml/serdes_gpx.cpp

```diff
@@ -283,6 +302,10 @@ void GpxParser::Pop(std::string_view tag)
     }
     ResetPoint();
   }
+
+  if (tag == gpx::kMetadata)
+    m_timestamp = base::INVALID_TIME_STAMP;
```

Comment:
Why is it reset here? Is `<metadata><time>` tag parsed and saved properly?

Reply:
The gpx parser works in another way than kml and Pop method is called on the closer tag (not opener).
Before exiting the parent tag we should clean up all the children's tags.

The `<time>` tags will always be parsed but when we met the `<metadata>` we should prepare the property for the next item.

We do not store the metadata's `<time>`.

We can solve this issue by skipping the time parsing in the:
```cpp
void GpxParser::ParseTimestamp(std::string const & value)
{
  if (prevTag == kMetadata)
    return;
  auto const timestamp = base::StringToTimestamp(value);
  if (timestamp == base::INVALID_TIME_STAMP)
    MYTHROW(kml::DeserializerGpx::DeserializeException, ("Failed to parse timestamp:", value));
  m_timestamp = std::move(timestamp);
}
```

or as already did in this PR

Reply:
TrackData has m_timestamp field that can be filled correctly from GPX, maybe we need another issue for it.

Not sure that I got the difference, our expat XML parser should always call Pop on closing tags. Push should be called for opener tags. How to investigate this issue?